### PR TITLE
Reorganize Tiered Storage docs

### PR DESCRIPTION
**Pull request by kbatuigas**
## Description

This PR reorganizes our set of Tiered Storage docs:
- improve the Tiered Storage index / landing page with a short description
- add Whole cluster restore and Remote read replicas to the Tiered Storage nav section
- Move Data Archiving from its own page to just a subsection in the main Tiered Storage doc
- Break out Fast commission and decommission as well as Remote recovery into their own docs, under the Tiered Storage nav section

Resolves https://github.com/redpanda-data/documentation-private/issues/2532
Resolves https://github.com/redpanda-data/documentation-private/issues/733
Resolves https://github.com/redpanda-data/documentation-private/issues/2533
Review deadline: 16 Aug 2024

## Page previews

[Tiered Storage landing page](https://deploy-preview-677--redpanda-docs-preview.netlify.app/current/manage/tiered-storage-linux/)
[Fast commission and decommission](https://deploy-preview-677--redpanda-docs-preview.netlify.app/current/manage/fast-commission-decommission/)
[Topic Recovery](https://deploy-preview-677--redpanda-docs-preview.netlify.app/current/manage/topic-recovery/)
[Tiered Storage (main doc) > Data archiving](https://deploy-preview-677--redpanda-docs-preview.netlify.app/current/manage/tiered-storage/#data-archiving)

## Checks

- [ ] New feature
- [ ] Content gap
- [ ] Support Follow-up
- [ ] Small fix (typos, links, copyedits, etc)

### CODE REVIEW

**Review by mattschumpert — APPROVED** [review]

File: modules/ROOT/nav.adoc [hunk 1]
```diff
@@ -161,9 +161,10 @@
 *** xref:manage:security/iam-roles.adoc[]
 ** xref:manage:tiered-storage-linux/index.adoc[Tiered Storage]
 *** xref:manage:tiered-storage.adoc[Tiered Storage]
```
Comment (asimms41) [inline modules/ROOT/nav.adoc]: Should this page be labelled differently to avoid having two pages with the same name.

File: modules/manage/partials/fast-commission-decommission.adoc [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+Tiered Storage can help make the cluster resize process faster and more cost-efficient by leveraging data that has already been uploaded to object storage. Instead of transferring all data from local storage to the reassigned replicas, the replicas can instead be initialized to rely more heavily on Tiered Storage for read requests.
+
+To accomplish this, Redpanda takes an on-demand snapshot at an offset that is already uploaded to object storage. This is a later offset compared to the default log start offset managed by glossterm:Raft[]. When an empty replica is initialized, the later offset uploaded to object storage can be used as the start offset source. Reads that access data before the uploaded offset can be executed as remote reads. Therefore, the only data that needs to be sent over the network to replicas is the data locally retained after the uploaded offset. This also results in less time taken overall to grow or shrink the cluster.
+
+== Configure fast commission and decommission
+
+To activate fast commission and decommission when brokers enter and leave the cluster:
+
+. Make sure to <<enable-tiered-storage,configure topics for Tiered Storage>>.
```
Comment (asimms41) [inline modules/manage/partials/fast-commission-decommission.adoc]:
```suggestion
. <<enable-tiered-storage,configure topics for Tiered Storage>>.
```
Also this text is a broken link.

File: modules/manage/partials/fast-commission-decommission.adoc [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+* `initial_retention_local_target_bytes_default`: Initial local retention size target for partitions of topics with Tiered Storage enabled. The default is null.
+** Use the `initial.retention.local.target.bytes` topic configuration property to override on the topic level.
+* `initial_retention_local_target_ms_default`: Initial local retention time target for partitions of topics with Tiered Storage enabled. The default is null.
+** Use the `initial.retention.local.target.ms` topic configuration property to override on the topic level.
+--
++
+If no values are set for the cluster configuration properties, all locally-retained data is delivered by default to the new 
+broker (learner) when joining a partition replica set.
+
+IMPORTANT: Because topics become more reliant on object storage to serve data, you may experience higher latency reads for data beyond the range of the configured local <<set-retention-limits,retention target>>. Carefully weigh the tradeoffs between faster broker commission/decommission and the increased read latency due to less data being available in local storage.
```
Comment (asimms41) [inline modules/manage/partials/fast-commission-decommission.adoc]: The retention target link doesn't work for me. Looks like a heading has been updated or no longer there.

File: modules/manage/partials/topic-recovery.adoc [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+When you create a topic, you can use remote recovery to download the topic data from object storage. This is useful when you need to restore a single topic in xref:manage:tiered-storage.adoc[Tiered Storage] that was accidentally deleted from a cluster.
+
+[IMPORTANT]
+====
+* Remote recovery is only safe when no other clusters are writing to the specified bucket or container.
+* You must enable `redpanda.remote.read` when using remote recovery.
+* If you later disable `redpanda.remote.read` on the remote recovery topic, previously hydrated (downloaded) data will not be used to serve requests.
+====
+
+To create a new topic using remote recovery (the recovered topic can read and write in the cloud):
```
Comment (asimms41) [inline modules/manage/partials/topic-recovery.adoc]: Could this be reworded to remove the brackets?

File: modules/manage/partials/topic-recovery.adoc [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+[,bash]
+----
+rpk topic create <topic_name> -c redpanda.remote.recovery=true -c redpanda.remote.write=true -c redpanda.remote.read=true
+----
+
+To create a new topic using remote recovery (while also disabling the `redpanda.remote.write` property):
```
Comment (asimms41) [inline modules/manage/partials/topic-recovery.adoc]: Could this be reworded to remove the brackets?

File: modules/deploy/pages/deployment-option/cloud/azure-private-link.adoc [hunk 6]
```diff
@@ -0,0 +1,20 @@
+= Configure Azure Private Link with the Cloud API
+:description: Set up Azure Private Link to securely access Redpanda Cloud.
+:page-cloud: true
+
+include::shared:partial$feature-flag.adoc[]
+
+The Redpanda Azure Private Link service provides secure access to Redpanda Cloud from your own VPC. Traffic over Azure Private Link does not go through the public internet because a Private Link connection is treated as its own private Azure service. While your VPC has access to the Redpanda VPC, Redpanda cannot access your VPC.
```
Comment (asimms41) [inline modules/deploy/pages/deployment-option/cloud/azure-private-link.adoc]: Is this page meant to be part of the PR? It doesn't seem related?

**Review by asimms41 — CHANGES_REQUESTED** [review]
Looks good. Just a few minor comments and maybe some unrelated content?

File: modules/deploy/pages/deployment-option/cloud/azure-private-link.adoc (hunk 6 above)
Comment (kbatuigas) [reply modules/deploy/pages/deployment-option/cloud/azure-private-link.adoc]: Correct, not supposed to be part of the PR. I'll try to clean it up

File: modules/manage/partials/tiered-storage.adoc [hunk 7]
```diff
@@ -1084,6 +1082,22 @@ Remote write uses a proportional derivative (PD) controller to minimize the back
...
 
+=== Data archiving
+
+If you only enable remote write on a topic, you have a simple backup to cloud storage that you can use for recovery. In the event of a data center failure, data corruption, or cluster migration, you can recover your archived data from the cloud back to your cluster.
+
+==== Configure data archiving
+
+Data archiving requires a <<set-up-tiered-storage,Tiered Storage configuration>>.
+
+To recover a topic from cloud storage, use xref:manage:topic-recovery.adoc[single topic recovery].
```
Comment (JakeSCahill) [inline modules/manage/partials/tiered-storage.adoc]: We agreed a while ago to standardize on 'object storage' instead of 'cloud storage'.

https://github.com/redpanda-data/documentation-private/issues/1317

File: modules/manage/pages/tiered-storage.adoc [hunk 8]
```diff
@@ -1,7 +1,7 @@
-= Tiered Storage
+= About Tiered Storage
```
Comment (JakeSCahill) [inline modules/manage/pages/tiered-storage.adoc]: This also contains 'how-to' info. I suggest giving this a title like 'Enable Tiered Storage'.

File: modules/ROOT/nav.adoc [hunk 9]
```diff
@@ -160,10 +163,11 @@
 **** xref:manage:security/console/tls-termination.adoc[TLS Termination]
 *** xref:manage:security/iam-roles.adoc[]
 ** xref:manage:tiered-storage-linux/index.adoc[Tiered Storage]
-*** xref:manage:tiered-storage.adoc[Tiered Storage]
-*** xref:manage:data-archiving.adoc[Data Archiving]
-** xref:manage:whole-cluster-restore.adoc[Whole Cluster Restore]
-** xref:manage:remote-read-replicas.adoc[Remote Read Replicas]
+*** xref:manage:tiered-storage.adoc[About Tiered Storage]
```
Comment (JakeSCahill) [inline modules/ROOT/nav.adoc]: This also contains 'how-to' info. I suggest giving this a title like 'Enable Tiered Storage'.

File: modules/manage/pages/tiered-storage.adoc [hunk 10]
```diff
@@ -1,7 +1,7 @@
...
 :description: Configure your Redpanda cluster to offload log segments to cloud storage and save storage costs.
 :page-context-links: [{"name": "Linux", "to": "manage:tiered-storage.adoc" },{"name": "Kubernetes", "to": "manage:kubernetes/storage/tiered-storage/k-tiered-storage.adoc" } ]
-:page-aliases: data-management:shadow-indexing.adoc, data-management:tiered-storage.adoc
+:page-aliases: data-management:shadow-indexing.adoc, data-management:tiered-storage.adoc, manage:data-archiving.adoc
 :page-categories: Management, High Availability, Data Replication
 :env-linux: true
```
Comment (micheleRP) [inline modules/manage/pages/tiered-storage.adoc:7]: @kbatuigas This page is somehow missing the admonition that it needs an enterprise license. It's noted in the prereqs section, but please add the admonition to the top of the page!

File: modules/ROOT/nav.adoc [hunk 11]
```diff
@@ -88,11 +88,14 @@
 ***** xref:manage:kubernetes/storage/k-persistent-storage.adoc[PersistentVolume]
 ***** xref:manage:kubernetes/storage/k-hostpath.adoc[hostPath]
 ***** xref:manage:kubernetes/storage/k-emptydir.adoc[emptyDir]
-**** xref:manage:kubernetes/tiered-storage-kubernetes/index.adoc[Tiered Storage]
-***** xref:manage:kubernetes/storage/tiered-storage/k-tiered-storage.adoc[Tiered Storage]
-***** xref:manage:kubernetes/storage/tiered-storage/k-data-archiving.adoc[Data Archiving]
 **** xref:manage:kubernetes/storage/k-resize-persistentvolumes.adoc[Expand PersistentVolumes]
 **** xref:manage:kubernetes/storage/k-delete-persistentvolume.adoc[Delete PersistentVolumes]
+*** xref:manage:kubernetes/tiered-storage/index.adoc[Tiered Storage]
```
Comment (asimms41) [inline modules/ROOT/nav.adoc:93]: It still feels a bit odd to have a section heading of Tiered Storage in the nav and then a duplicate sub-section heading underneath.

Comment (JakeSCahill) [reply modules/ROOT/nav.adoc:93]: Where's the duplicate?

File: modules/manage/partials/remote-read-replicas.adoc [hunk 12]
```diff
@@ -9,7 +9,7 @@ ifdef::env-kubernetes[]
 endif::[]
 ifndef::env-kubernetes[]
 :tiered-storage-link: manage:tiered-storage.adoc
-:data-archiving-link: manage:data-archiving.adoc
+:data-archiving-link: manage:tiered-storage.adoc#data-archiving
```
Comment (asimms41) [inline modules/manage/partials/remote-read-replicas.adoc:12]: paragraphs below talk about cloud object storage not object storage. Is this correct?

File: modules/ROOT/nav.adoc (hunk 11 above)
Comment (asimms41) [reply modules/ROOT/nav.adoc:93]: It's been updated. It now reads Tiered Storage > Use Tiered Storage.

File: modules/ROOT/nav.adoc [hunk 13]
```diff
@@ -112,7 +115,7 @@
 *** xref:manage:kubernetes/k-manage-resources.adoc[Manage Pod Resources]
 *** xref:manage:kubernetes/k-scale-redpanda.adoc[Scale]
 *** xref:manage:kubernetes/k-decommission-brokers.adoc[Decommission Brokers]
-*** xref:manage:kubernetes/k-whole-cluster-restore.adoc[Whole Cluster Restore]
+*** xref:[Whole Cluster Restore]
```
Comment (asimms41) [inline modules/ROOT/nav.adoc]: This is showing as a broken/unresolved link in the nav tree.

File: modules/manage/partials/whole-cluster-restore.adoc [hunk 14]
```diff
@@ -15,7 +15,7 @@ NOTE: You cannot use whole cluster restore if the target cluster is in xref:{lin
 ====
 Whole cluster restore is not a fully-functional disaster recovery solution. It does not provide snapshot-style consistency. Some partitions in some topics will be more up-to-date than others. Committed transactions are not guaranteed to be atomic.
 
-If you need to restore only a subset of topic data, consider using xref:{link-tiered-storage}#remote-recovery[remote recovery] instead of a whole cluster restore.
+If you need to restore only a subset of topic data, consider using xref:manage:topic-recovery.adoc[topic recovery] instead of a whole cluster restore.
```
Comment (asimms41) [inline modules/manage/partials/whole-cluster-restore.adoc:18]: Line 51 - might be a separate issue, but it looks like the cluster property has moved.

File: modules/manage/partials/remote-read-replicas.adoc (hunk 12 above)
Comment (kbatuigas) [reply modules/manage/partials/remote-read-replicas.adoc:12]: @asimms41 I can go ahead and change it 👍

File: modules/manage/partials/whole-cluster-restore.adoc (hunk 14 above)
Comment (kbatuigas) [reply modules/manage/partials/whole-cluster-restore.adoc:18]: @asimms41 ah yes, we moved object storage (tiered storage) properties to a [new page](https://deploy-preview-677--redpanda-docs-preview.netlify.app/current/reference/properties/object-storage-properties/). I'll fix this one but I'm not sure if we have an issue yet to fix all the old links from cluster-properties.adoc. I'll ask Paulo.

File: modules/manage/partials/tiered-storage.adoc [hunk 15]
```diff
@@ -5,6 +5,11 @@ ifndef::env-kubernetes[]
 :data-archiving-link: manage:data-archiving.adoc
 endif::[]
 
+[NOTE]
+====
+include::shared:partial$enterprise-license.adoc[]
```
Comment (asimms41) [inline modules/manage/partials/tiered-storage.adoc:10]: Line 855. The cluster properties are not linking to the same named properties.

File: modules/manage/partials/tiered-storage.adoc [hunk 16]
```diff
@@ -1084,6 +1087,22 @@ Remote write uses a proportional derivative (PD) controller to minimize the back
...
 
+=== Data archiving
+
+If you only enable remote write on a topic, you have a simple backup to cloud storage that you can use for recovery. In the event of a data center failure, data corruption, or cluster migration, you can recover your archived data from the cloud back to your cluster.
+
+==== Configure data archiving
+
+Data archiving requires a <<set-up-tiered-storage,Tiered Storage configuration>>.
+
+To recover a topic from object storage, use xref:manage:topic-recovery.adoc[single topic recovery].
```
Comment (asimms41) [inline modules/manage/partials/tiered-storage.adoc]: In Kubernetes version of this link, I'm taken to the Linux content.

File: modules/manage/partials/tiered-storage.adoc [hunk 17]
```diff
@@ -1106,7 +1125,7 @@ To enable remote read on an existing topic:
 rpk topic alter-config <topic_name> --set redpanda.remote.read=true
 ----
 
-See also: xref:manage:kubernetes/k-remote-read-replicas.adoc[Remote Read Replicas]
+See also: xref:manage:topic-recovery.adoc[Topic Recovery], xref:manage:kubernetes/k-remote-read-replicas.adoc[Remote Read Replicas]
```
Comment (asimms41) [inline modules/manage/partials/tiered-storage.adoc]: Topic recovery link goes to Linux version?

File: modules/manage/partials/remote-read-replicas.adoc [hunk 18]
```diff
@@ -9,7 +9,7 @@ ifdef::env-kubernetes[]
...
 endif::[]
 
 A Remote Read Replica topic is a read-only topic that mirrors a topic on a different cluster. Remote Read Replicas work with both xref:{tiered-storage-link}[Tiered Storage] and xref:{data-archiving-link}[archival storage].
```
Comment (asimms41) [inline modules/manage/partials/remote-read-replicas.adoc:15]: For Kubernetes, the second link goes to a section outside of Kubernetes. Is that correct?

**Review by asimms41 — DISMISSED** [review]
Looks great. Just a few suggestions.

**Review by asimms41 — COMMENTED** [review]
Just minor thing not related to the changes in this PR. This link appears in a number of places but it looks like heading has been removed [Local storage housekeeping](https://deploy-preview-677--redpanda-docs-preview.netlify.app/current/manage/cluster-maintenance/disk-utilization/#local-storage-housekeeping)

File: modules/manage/pages/security/iam-roles.adoc [hunk 19]
```diff
@@ -128,7 +128,7 @@ If you are using Microsoft Azure as your cloud provider, you must satisfy the fo
 
 == Configuring IAM roles
 
-After satisfying the prerequisites for your cloud platform, edit the Redpanda cluster configuration by running `rpk cluster config edit`. Set the xref:reference:cluster-properties.adoc#cloud_storage_credentials_source[`cloud_storage_credentials_source`] property to the appropriate value for your use case. The following table shows all possible values and their descriptions.
+After satisfying the prerequisites for your cloud platform, edit the Redpanda cluster configuration by running `rpk cluster config edit`. Set the xref:reference:object-storage-properties.adoc#cloud_storage_credentials_source[`cloud_storage_credentials_source`] property to the appropriate value for your use case. The following table shows all possible values and their descriptions.
```
Comment (JakeSCahill) [inline modules/manage/pages/security/iam-roles.adoc]:
```suggestion
After satisfying the prerequisites for your cloud platform, edit the Redpanda cluster configuration by running `rpk cluster config edit`. Set the xref:reference:properties/object-storage-properties.adoc#cloud_storage_credentials_source[`cloud_storage_credentials_source`] property to the appropriate value for your use case. The following table shows all possible values and their descriptions.
```

File: modules/manage/pages/tiered-storage.adoc (hunk 10 above)
Comment (Feediver1) [reply modules/manage/pages/tiered-storage.adoc:7]: +1

**Review by JakeSCahill — APPROVED** [review]
